core/templates/services/audio-player.service.ts

```diff
@@ -16,162 +16,235 @@
-        if (_currentTrack && _currentTrack.audio) {
-          return Math.round(_currentTrack.audio.duration);
-        } else {
-          return 0;
-        }
-      },
-      getProgress: function() {
-        if (!_currentTrack) {
-          return 0;
-        }
-        return _currentTrack.progress;
-      },
-      setProgress: function(progress) {
-        if (_currentTrack) {
-          _currentTrack.progress = progress;
-        }
-      },
-      isPlaying: function() {
-        return Boolean(_currentTrack && !_currentTrack.paused);
-      },
-      isTrackLoaded: function() {
-        return Boolean(_currentTrack);
-      },
-      clear: function() {
-        _currentTrack = null;
-        _currentTrackFilename = null;
-      },
-      get onAutoplayAudio() {
-        return _autoplayAudioEventEmitter;
+  load(filename: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this._load(filename, resolve, reject);
+    });
+  }
+
+  play(): void {
+    this._play();
+  }
+
+  pause(): void {
+    this._pause();
+  }
+
+  stop(): void {
+    this._stop();
+  }
+
+  rewind(seconds: number): void {
+    this._rewind(seconds);
+  }
+
+  forward(seconds: number): void {
+    this._forward(seconds);
+  }
+
+  getCurrentTime(): number {
+    if (this._currentTrack) {
+      const sec = this._currentTrack.seek();
+      if (typeof sec !== 'number') {
+        return 0;
       }
-    };
+      return Math.floor(sec);
+    } else {
+      return 0;
+    }
+  }
+
+  setCurrentTime(val: number): void {
+    if (!this._currentTrack) {
+      return;
+    }
+    if (val < 0) {
+      this._currentTrack.seek(0);
+      return;
+    }
+    if (val > this._currentTrack.duration()) {
+      this._currentTrack.seek(this._currentTrack.duration());
+      return;
+    }
+    this._lastPausePos = val;
+    this._currentTrack.seek(Math.floor(val));
+  }
+
+  getAudioDuration(): number {
+    if (this._currentTrack) {
+      return this._currentTrack.duration();
+    } else {
+      return 0;
+    }
+  }
+
+  getProgress(): number {
+    if (!this._currentTrack) {
+      return 0;
+    }
+    return (
+      this._currentTrack.seek() as number) / this._currentTrack.duration();
+  }
+
+  setProgress(progress: number): void {
+    if (progress < 0 || progress > 1) {
+      return;
+    }
+    if (this._currentTrack) {
+      this._currentTrack.seek(progress * this._currentTrack.duration());
+    }
   }
-]);
+
+  isPlaying(): boolean {
+    return this._currentTrack && this._currentTrack.playing();
+  }
+
+  isTrackLoaded(): boolean {
+    return this._currentTrack !== null;
+  }
+
+  clear(): void {
+    if (this._currentTrack && this.isPlaying()) {
+      this._currentTrack.stop();
+    }
+    this._currentTrack = null;
+  }
+
+  get currentTime(): number {
+    return this.getCurrentTime();
+  }
+
+  get viewUpdate(): EventEmitter<void> {
+    return this._updateViewEventEmitter;
+  }
+
+  get onAutoplayAudio(): EventEmitter<void> {
```

Comment:
is this used anywhere?

Reply:
[optional] Can you please do a quick check and remove this function if not needed?